### ISSUE TITLE
[BUG]: broken build [-Wincompatible-pointer-types]

### ISSUE THREAD

**Issue by remicollet** [phalcon/cphalcon#16531]
With recent GCC 14 (on Fedora 40)

Build fails with tons of:

```
/builddir/build/BUILD/php83-php-phalcon5-5.6.1/phalcon-5.6.1/phalcon.zep.c: In function 'zim_Phalcon_Storage_Adapter_Weak_increment':
/opt/remi/php83/root/usr/include/php/Zend/zend_API.h:2059:58: error: passing argument 2 of 'zend_parse_arg_str' from incompatible pointer type [-Wincompatible-pointer-types]
 2059 |                 if (UNEXPECTED(!zend_parse_arg_str(_arg, &dest, check_null, _i))) { \
/opt/remi/php83/root/usr/include/php/Zend/zend_portability.h:363:52: note: in definition of macro 'UNEXPECTED'
  363 | # define UNEXPECTED(condition) __builtin_expect(!!(condition), 0)
      |                                                    ^~~~~~~~~
/opt/remi/php83/root/usr/include/php/Zend/zend_API.h:2066:9: note: in expansion of macro 'Z_PARAM_STR_EX'
 2066 |         Z_PARAM_STR_EX(dest, 0, 0)
      |         ^~~~~~~~~~~~~~
/builddir/build/BUILD/php83-php-phalcon5-5.6.1/phalcon-5.6.1/phalcon.zep.c:204823:17: note: in expansion of macro 'Z_PARAM_STR'
204823 |                 Z_PARAM_STR(key)
       |                 ^~~~~~~~~~~
/opt/remi/php83/root/usr/include/php/Zend/zend_API.h:2241:76: note: expected 'zend_string **' {aka 'struct _zend_string **'} but argument is of type 'zval *' {aka 'struct _zval_struct *'}
 2241 | static zend_always_inline bool zend_parse_arg_str(zval *arg, zend_string **dest, bool check_null, uint32_t arg_num)
      |                                                              ~~~~~~~~~~~~~~^~~~
```

Was already reported for a long time.

This was a warning, this in now an error

**Comment by Jeckerson** [comment]
Yes, I remember your report and issue is still opened - https://github.com/zephir-lang/zephir/issues/2326

There was my attempts during last months to improve it, but it is not trivial task to do. As whole Zephir relies on `zval` instead of specific types as `zend_string` or any other. But, there are already improvements in another types, such as `zend_bool` and `zend_long`. Which are located in Zephir's `development` branch.

I'll try to review it in near future...

**Comment by stfast** [comment]
Fedora 40 was released a week ago and we would go along with PHP 8.3 in this release.
Are there any timeline to compile phalcon with GCC 14, so we can plan system upgrade?

**Comment by remicollet** [comment]
@stfast notice than on Fedora, you can use my repo which provides more PHP versions and more extensions than in official repository, including phalcon, see https://rpms.remirepo.net/wizard/

**Comment by stfast** [comment]
Remi I admire your work, great, hard and wonderfull job you you are working, but I've allways had problems with (cycylic) package dependancies with your Remi repos on Fedora upgrades in the past. Thank you reminding me, I will try.

**Comment by andypost** [comment]
Any progress on the issue, I have to remove the package from Alpinelinux repo as no way to build it using GCC14

**Comment by HiralRajgor** [comment]
@niden 

Please help us on this issue, as alpine linux package is removed from repo due to this.

**Comment by Jeckerson** [comment]
On it

**Comment by jukuan** [comment]
same issue. Any updates on it? or replacements?suggestions?
A lot of people really need the phalcon package for alpine :)
Thanks in advance.

**Comment by Jeckerson** [comment]
@andypost @jukuan How do you install in Alpine?

**Comment by Jeckerson** [comment]
Just tried to compile on latest version of alpine with PHP8.3 and it requires near 5Gb of RAM:
![image](https://github.com/user-attachments/assets/ca3b6072-5655-40de-a0d1-b9f59ea71070)

before that it killed process with error:
```
cc: fatal error: Killed signal terminated program cc1
```

For some urgent cases: use swap files with 4-5Gb to compile it on VMs/VPS with small RAM.
Guide how to create swap file - [Create swap file on Linux](https://gist.github.com/Jeckerson/2322849d9f5dbbc9543c99e49341177e)

I'll investigate what might be possible to do, but current warnings **are not** the core problem is original issue.

**Comment by HiralRajgor** [comment]
> same issue. Any updates on it? or replacements?suggestions? A lot of people really need the phalcon package for alpine :) Thanks in advance.

Yeah True, I had to work on weekend, as my prod release got delayed due to package removal lol.

But I did some alternates, it might not be the best way but anyways it saved me :

- I already have the compiled .so file downloaded from my running container **phalcon.so** so I have added this file in my project, and when my docker container runs, phalcon.so file is available at [https://github.com/phalcon/cphalcon/releases/tag/v5.8.0] as well.
i have just copied this phalcon.so file to /usr/lib/php82/modules in my alpine linux
- Then created one phalcon.ini file /etc/php82/conf.d/phalcon.ini, this file is containing **extension=phalcon**
- I found one more phalcon file in my running container and downloaded this file and added in my project, **php_phalcon.h** and then same when docker container runs copied this file to /usr/include/php82/ext/phalcon/php_phalcon.h

**Comment by jukuan** [comment]
My solution is to build the Phalcon from the source code for my alpine container. The basic Dockerfile snippes is below.  Don't rely on the testing repositories any more :)

Feel free to use it and/or tell me what might be improved.

```
FROM php:8.2-cli-alpine

ARG PHALCON_VER=5.8.0

RUN set -eux; \
    cd /tmp && \
    curl -sSJL -o phalcon.tar.gz "https://github.com/phalcon/cphalcon/archive/v${PHALCON_VER}.tar.gz" && \
    mkdir -p /usr/src/phalcon && \
    tar -xzf phalcon.tar.gz -C /usr/src/phalcon --strip-components=1 && \
    cd /usr/src/phalcon/build && ./install && \
    docker-php-ext-enable decimal phalcon  && \
    \
# Clean up
    apk del \
        autoconf \
        gcc \
        g++ \
        git \
        libaio \
        libc-dev \
        make \
        pkgconf \
        re2c \
    && rm -rf /var/cache/apk/* /tmp/*
    
COPY --from=composer:2.8 /usr/bin/composer /usr/local/bin/composer

WORKDIR /var/www/html
CMD ["php", "-a"]
```

**Comment by Jeckerson** [comment]
Current temporary solution if it fails to compile:

```bash
CFLAGS="-Wno-error=incompatible-pointer-types" pecl install phalcon
# or to silence:
CFLAGS="-Wno-incompatible-pointer-types" pecl install phalcon
```

**Comment by Jeckerson** [comment]
Fixed since Zephir [0.20.0](https://github.com/zephir-lang/zephir/releases/tag/0.20.0).
